google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/project/ProjectSelector.java

```diff
@@ -0,0 +1,220 @@
+  }
+
+  private void createUIComponents() {
+    // layout - in the center of panel, horizontal line of project/account labels in a panel
+    // ends with a fixed size browse button.
+    setLayout(new BorderLayout());
+
+    JPanel staticInfoPanel = new JPanel();
+    staticInfoPanel.setLayout(new FlowLayout(FlowLayout.LEFT, 5, 2));
+    staticInfoPanel.setBorder(UIManager.getBorder("TextField.border"));
+
+    projectNameLabel = new HyperlinkLabelWithStateAccess();
+    projectNameLabel.addHyperlinkListener(
+        (event) -> {
+          if (event.getEventType() == EventType.ACTIVATED) {
+            handleOpenProjectSelectionDialog();
+          }
+        });
+    staticInfoPanel.add(projectNameLabel);
+
+    projectAccountSeparatorLabel = new JBLabel("/");
+    projectAccountSeparatorLabel.setVisible(false /* only visible when project is selected. */);
+    staticInfoPanel.add(projectAccountSeparatorLabel);
+
+    accountInfoLabel = new HyperlinkLabelWithStateAccess();
+    accountInfoLabel.addHyperlinkListener(
+        (event) -> {
+          if (event.getEventType() == EventType.ACTIVATED) {
+            handleOpenProjectSelectionDialog();
+          }
+        });
+    staticInfoPanel.add(accountInfoLabel);
+
+    ComponentWithBrowseButton<JPanel> componentWithBrowseButton =
+        new ComponentWithBrowseButton<>(
+            staticInfoPanel, (actionEvent) -> handleOpenProjectSelectionDialog());
+
+    add(componentWithBrowseButton);
+  }
+
+  @VisibleForTesting
+  void handleOpenProjectSelectionDialog() {
+    ProjectSelectionDialog projectSelectionDialog = createProjectSelectionDialog(this);
+    CloudProject newSelection = projectSelectionDialog.showDialog(cloudProject);
+
+    // if null, it means no change or user cancelled selection dialog - no update required.
+    if (newSelection != null) {
+      setSelectedProject(newSelection);
+      notifyProjectSelectionListeners();
+    }
+  }
+
+  private void updateProjectAndUserInformation(CloudProject selection) {
+    if (selection == null || Strings.isNullOrEmpty(selection.getProjectName())) {
+      projectNameLabel.setHyperlinkText(
+          GctBundle.getString("project.selector.no.selected.project"));
+      accountInfoLabel.setHyperlinkText("");
+      accountInfoLabel.setIcon(null);
+      projectAccountSeparatorLabel.setVisible(false);
+
+    } else {
+      projectNameLabel.setHyperlinkText(selection.getProjectName());
+      projectAccountSeparatorLabel.setVisible(true);
+      // first just show account email, then expand with name/picture if this account is signed in.
+      accountInfoLabel.setHyperlinkText(selection.getGoogleUsername());
+      Optional<CredentialedUser> loggedInUser =
+          Services.getLoginService().getLoggedInUser(selection.getGoogleUsername());
+      if (loggedInUser.isPresent()) {
+        accountInfoLabel.setHyperlinkText(
+            loggedInUser.get().getName() + " (" + loggedInUser.get().getEmail() + ")");
```

Comment:
This might be best expressed via a format string.

Reply:
great idea.